prompt_toolkit/buffer.py

```diff
@@ -1405,14 +1405,40 @@ def open_in_editor(self, validate_and_handle: bool = False) -> 'asyncio.Task[Non
             raise EditReadOnlyBuffer()
 
         # Write to temporary file
-        if self.tempfile:
+        try:
```

Comment:
This function is becoming long. Is there any way to break it up in maybe two or three functions. (It is hard to review).

Reply:
Split tempfile creation into simple vs complex.